### PR TITLE
feat: align file trigger syntax with class trigger

### LINKED ISSUE #7928: Depreciate `Parse.Cloud.beforeSaveFile(() => {})` in preference to `Parse.Cloud.beforeSave(Parse.File, () => {})`
### New Feature / Enhancement Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

### Current Limitation

Currently, the file triggers have their own unique syntax, which isn't necessary as they follow the standard trigger types (beforeSave, afterSave, etc).

### Feature / Enhancement Description

Allow `Parse.Cloud.beforeSave(Parse.File)` and depreciate existing file methods.

### Example Use Case

n/a

### Alternatives / Workarounds

n/a

### 3rd Party References

n/a

### PR DESCRIPTION
**Pull request by dblythy**
### New Pull Request Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am creating this PR in reference to an [issue](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

### Issue Description

Currently, File Triggers have their own unique syntax. This is unnecessary, and they should follow the same syntax for other file triggers.

Related issue: #7928

### Approach

Add new file trigger method, depreciate old syntax to be removed in V7.

### TODOs before merging

- [x] Add tests
- [ ] Add changes to documentation (guides, repository pages, in-code descriptions)
- [x] A changelog entry is created automatically using the pull request title (do not manually add a changelog entry)

### CODE REVIEW

**Comment by mtrezza** [comment]
Could you give this PR a more descriptive title?

File: DEPRECATIONS.md
Comment (mtrezza) [inline DEPRECATIONS.md]: From reading this I'm not sure the majority will understand what this is about? "in reference to"?

File: src/cloud-code/Parse.Cloud.js
Comment (mtrezza) [inline src/cloud-code/Parse.Cloud.js]: Could you change to:

```js
solution: 'Use Parse.Cloud.beforeSave(Parse.File, (request) -> {}) instead.',
```

Also in the other messages.

File: src/triggers.js
Comment (mtrezza) [inline src/triggers.js:954]: I think I overlooked it, but what is the `FileClassName`?

**Review by mtrezza — CHANGES_REQUESTED** [review]

File: src/triggers.js
Comment (dblythy) [reply src/triggers.js:954]: https://github.com/parse-community/parse-server/blob/48bd512eeb47666967dff8c5e723ddc5b7801daa/src/triggers.js#L24

**Comment by mtrezza** [comment]
We shouldn't use code to explain code changes - could you just explain the PR in prose form in the title?

**Review by Moumouls — CHANGES_REQUESTED** [review]
LGTM @dblythy 

I'm just blocking the PR because of the deprecation, we should have the docs PR ready also.
Then both could be merged at the same time.

Feel free @dblythy to link the docs pr here 👍

File: DEPRECATIONS.md
Comment (mtrezza) [inline DEPRECATIONS.md]: I resolved the DEP ID conflict, DEPPS5 was already in use, do we have any other ID references that need to be updated?

Comment (dblythy) [reply DEPRECATIONS.md]: Nope! 😊

**Comment by mtrezza** [comment]
Is this PR ready for merge?

File: src/cloud-code/Parse.Cloud.js
Comment (mtrezza) [inline src/cloud-code/Parse.Cloud.js:495]: This removes the API reference for deprecated triggers. I think they should be kept but flagged as deprecated. There is a JSDoc flag for that I believe. Otherwise the API will be undocumented.

Also, the description in existing triggers should be amended, because they only mention "Class" but it may not be obvious that Parse.File is also just another class in that sense.

**Review by mtrezza — CHANGES_REQUESTED** [review]

File: DEPRECATIONS.md
Comment (mtrezza) [inline DEPRECATIONS.md]: This will likely have to change to `DEPPS7`, I will merge https://github.com/parse-community/parse-server/pull/7953 first.

**Review by mtrezza — CHANGES_REQUESTED** [review]

File: DEPRECATIONS.md
Comment (mtrezza) [reply DEPRECATIONS.md]: Did it myself

**Review by mtrezza — APPROVED** [review]
Looks good! I think we can merge this, as the deprecation warning is self-descriptive and the API reference should indicate what is deprecated and which syntax should be used instead.

There is still time to write the docs PR. The docs will be updated on stable release which is planned for in ~1 month when this feature goes out of beta phase.

**Comment by dblythy** [comment]
I'd be happy to do the docs

**Comment by mtrezza** [comment]
That'd be great! Re-running the failed tests; once they pass we can merge.

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.3.0-alpha.14](https://github.com/parse-community/parse-server/releases/tag/5.3.0-alpha.14)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.3.0-beta.1](https://github.com/parse-community/parse-server/releases/tag/5.3.0-beta.1)

**Comment by mtrezza** [comment]
@dblythy Do you have a docs PR for that? This will go into stable release in few days.

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.3.0](https://github.com/parse-community/parse-server/releases/tag/5.3.0)

**Pull request merged**
